server/modules/nodeDef/repository/nodeDefRepository.js

```diff
@@ -255,6 +255,16 @@ export const updateNodeDefAnalysisCycles = async (surveyId, cycleKeys, client =
     [cycleKeys]
   )
 
+export const updateNodeDefAnalysisScript = async ({ surveyId, script, nodeDef }, client = DB) =>
```

Comment:
can't you use NodeDefRepository.updateNodeDefProps for this?

Reply:
I need to pass all of this

```
export const updateNodeDefProps = async (
  user,
  surveyId,
  nodeDefUuid,
  parentUuid,
  props,
  propsAdvanced = {},
  system = false,
  client = db
) =><
````